WebAssembly/Instructions/Int32TruncateSaturateFloat32Signed.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System.Reflection.Emit;
+using WebAssembly.Runtime;
+using WebAssembly.Runtime.Compilation;
+
+namespace WebAssembly.Instructions
+{
+    /// <summary>
+    /// Truncate (with saturation) a 32-bit float to a signed 32-bit integer.
+    /// </summary>
+    public class Int32TruncateSaturateFloat32Signed : MiscellaneousInstruction
+    {
+        /// <summary>
+        /// Always <see cref="MiscellaneousOpCode.Int32TruncateSaturateFloat32Signed"/>.
+        /// </summary>
+        public override MiscellaneousOpCode MiscellaneousOpCode =>
+            MiscellaneousOpCode.Int32TruncateSaturateFloat32Signed;
+
+        /// <summary>
+        /// Creates a new  <see cref="Int32TruncateSaturateFloat32Signed"/> instance.
+        /// </summary>
+        public Int32TruncateSaturateFloat32Signed()
+        {
+        }
+
+        internal sealed override void Compile(CompilationContext context)
+        {
+            var stack = context.Stack;
+            if (stack.Count == 0)
+                throw new StackTooSmallException(MiscellaneousOpCode.Int32TruncateSaturateFloat32Signed, 1, 0);
+
+            var type = stack.Pop();
+            if (type != WebAssemblyValueType.Float32)
+                throw new StackTypeInvalidException(MiscellaneousOpCode.Int32TruncateSaturateFloat32Signed, WebAssemblyValueType.Float32, type);
+
+            context.Emit(OpCodes.Call, context[HelperMethod.Int32TruncateSaturateFloat32Signed, (helper, c) =>
+            {
+                var builder = c.CheckedExportsBuilder.DefineMethod(
+                    "☣ Int32TruncateSaturateFloat32Signed",
+                    CompilationContext.HelperMethodAttributes,
+                    typeof(int),
+                    new[] { typeof(float) }
+                );
+
+                var il = builder.GetILGenerator();
+
+                var label1 = il.DefineLabel();
+                var label2 = il.DefineLabel();
+                var label3 = il.DefineLabel();
```

Comment:
Are there better names that could be used for these labels? (Same question in other instances, but only asking here.)

Reply:
Yes, good idea - I've done that.